### PR TITLE
BUG: Dont set backend on macOS

### LINKED ISSUE #2560: BUG: nilearn.plotting forces matplotlib backend change to Agg on macOS
on macOS:
```
>>> import matplotlib
>>> assert matplotlib.get_backend() == 'Qt5Agg'
>>> import nilearn.plotting
>>> assert matplotlib.get_backend() == 'Qt5Agg', matplotlib.get_backend()
AssertionError: agg
```
I have `backend : Qt5Agg\ninteractive : True` in `~/.matplotlib/matplotlibrc`. I assume it's due to this line:

https://github.com/nilearn/nilearn/blob/master/nilearn/plotting/__init__.py#L37

It seems like unwanted behavior that the backend is changed.

### PR DESCRIPTION
**Pull request by larsoner**
Closes #2560

### CODE REVIEW

File: nilearn/plotting/__init__.py [hunk 1]
```diff
@@ -30,12 +30,12 @@ def _set_mpl_backend():
         if 'inline' in current_backend or 'nbagg' in current_backend:
             return
         # Set the backend to a non-interactive one for unices without X
-        if (os.name == 'posix' and 'DISPLAY' not in os.environ
-                and not (sys.platform == 'darwin'
-                         and 'macosx' in current_backend)
-        ):
+        # (see gh-2560)
+        if (sys.platform not in ('darwin', 'win32') and
+                'DISPLAY' not in os.environ):
```
Comment (agramfort) [inline nilearn/plotting/__init__.py:35]: this changes a bit the logic as you assume that a mac will not be used as a remove headless server but i agree it's unlikely

I can live with that

Comment (larsoner) [reply nilearn/plotting/__init__.py:35]: Related discussion is in #2560

FWIW I think/assume Travis and GitHub macOS CIs are headless and they gracefully fall back to software rendering anyway.

**Comment by agramfort** [comment]
this solution is ok for  me but maybe DISPLAY is not defined on travis as
xquartz may not be installed.

**Comment by larsoner** [comment]
> this solution is ok for  me but maybe DISPLAY is not defined on travis as xquartz may not be installed.

I don't see why this is a problem based on the conditional above. If the sys is macOS/darwin, then the presence of the DISPLAY variable is irrelevant (the second part of the conditional isn't evaluated)

**Comment by jeromedockes** [comment]
To me it looks like matplotlib already selects an appropriate backend, starting here:
https://github.com/matplotlib/matplotlib/blob/f7ae15780857c9ddedd09dd614b68069bcb668cb/lib/matplotlib/pyplot.py#L234

also, to find out which interactive framework is available they seem to use `cbook._get_running_interactive_framework()`

btw, I think all the non-interactive backends ie pdf, ps, svg, cairo ... should be left alone and not just nbagg and inline

**Review by agramfort — APPROVED** [review]

**Comment by bthirion** [comment]
Have we reached a consensus ? If everybody's happy, I'll merge tomorrow.

**Comment by thomasbazeille** [comment]
Seems ready for merge @bthirion

**Comment by bthirion** [comment]
Thx !

**Comment by jeromedockes** [comment]
Can someone reproduce the problem where matplotlib chooses a wrong
backend on a headless machine, which makes it necessary for nilearn to
set the backend?

**Pull request merged**